src/xb-database-manager.c

```diff
@@ -552,13 +554,85 @@ create_empty_query_results (void)
   JsonObject *object;
 
   object = json_object_new ();
-  json_object_set_int_member (object, RESULTS_MEMBER_NUM_RESULTS, 0);
-  json_object_set_int_member (object, RESULTS_MEMBER_OFFSET, 0);
-  json_object_set_array_member (object, RESULTS_MEMBER_RESULTS, json_array_new ());
+  json_object_set_int_member (object, QUERY_RESULTS_MEMBER_NUM_RESULTS, 0);
+  json_object_set_int_member (object, QUERY_RESULTS_MEMBER_OFFSET, 0);
+  json_object_set_array_member (object, QUERY_RESULTS_MEMBER_RESULTS, json_array_new ());
 
   return object;
 }
 
+static JsonObject *
+xb_database_manager_fix_query_internal (XbDatabaseManager *self,
+                           DatabasePayload *payload,
+                           GHashTable *query_options,
+                           GError **error_out)
+{
+  gchar *spell_corrected_query_str = NULL, *no_stop_words = NULL;
+  gchar **words = NULL, **words_iter = NULL;
+  gchar **filtered_words = NULL, **filtered_iter = NULL;
+  GError *error = NULL;
+  const gchar *query_str;
+  const gchar *match_all;
+  XapianStopper *stopper;
+  JsonObject *retval;
+
+  retval = json_object_new ();
+
+  query_str = g_hash_table_lookup (query_options, QUERY_PARAM_QUERYSTR);
+  match_all = g_hash_table_lookup (query_options, QUERY_PARAM_MATCH_ALL);
+
+  stopper = xapian_query_parser_get_stopper (payload->qp);
+
+  if (query_str == NULL || match_all != NULL)
+    {
+      g_set_error (error_out, XB_ERROR,
+                  XB_ERROR_INVALID_PARAMS,
+                  "Query parameter must be set, and must not be match all.");
+      goto out;
+    }
+  else if (stopper != NULL)
+    {
+      filtered_words = g_new0 (gchar *, g_strv_length (words) + 1);
+      words = g_strsplit (query_str, " ", -1);
+
+      filtered_iter = filtered_words;
+      words_iter = words;
+
+      for (; *words_iter != NULL; words_iter++)
```

Comment:
You can move `words_iter = words` before the first semi to be a little more concise